File: demo/src/main/java/com/baozi/demo/item/cart/CartBean.java
package com.baozi.demo.item.cart;

import com.baozi.treerecyclerview.annotation.TreeDataType;


@TreeDataType(bindField = "type")
public class CartBean {
    int childSum; //子条目数量
    public int type = 1;

    public CartBean(int childSum) {
        this.childSum = childSum;
    }

    @TreeDataType(iClass = CartGroupItem2.class)
    public static class CartBean2 {
        int childSum; //子条目数量

        public CartBean2(int childSum) {
            this.childSum = childSum;
        }

        @TreeDataType(iClass = CartGroupItem3.class)
        public static class CartBean3 {
            int childSum; //子条目数量

            public CartBean3(int childSum) {
                this.childSum = childSum;
            }
        }
    }
}

File: demo/src/main/java/com/baozi/demo/item/cart/CartBean.kt
package com.baozi.demo.item.cart

import com.baozi.treerecyclerview.annotation.TreeDataType


@TreeDataType(bindField = "type")
class CartBean(internal var childSum: Int //子条目数量
) {
    @JvmField
    var type = 1

    @TreeDataType(iClass = CartGroupItem2::class)
    class CartBean2(internal var childSum: Int //子条目数量
    ) {

        @TreeDataType(iClass = CartGroupItem3::class)
        class CartBean3(internal var childSum: Int //子条目数量
        )
    }
}
